src/main/java/org/craftercms/engine/controller/rest/MonitoringController.java

```diff
@@ -56,4 +69,29 @@ public List<Map<String, Object>> getLoggedEvents(@RequestParam @ValidSiteId Stri
         return CircularQueueLogAppender.getLoggedEvents(site, since);
     }
 
+    @Override
+    @GetMapping(ROOT_URL + STATUS_URL)
+    public StatusInfo getCurrentStatus(@RequestParam(name = "site", required = false) String site,
+                                       @RequestParam(name = "token") String token)
+            throws InvalidManagementTokenException {
+        validateToken(token);
+
+        if (site != null) {
+            String paramNameValidationKey = SITE_ID.typeKey;
+            try {
+                validator.getValidInput(paramNameValidationKey, site, paramNameValidationKey, MAXIMUM_SITE_ID_KEY_LENGTH, false);
+            } catch (ValidationException e) {
+                throw new SiteContextInvalidException(format("Invalid site Id: '%s'.", site));
```

Comment:
Will this return a 500?
I would expect a 400 in the case the actual site ID is an invalid value.

cc @sumerjabri @avasquez614

Reply:
Updated to return 400 if the actual site ID is an invalid one. Thanks!